ElemType-only patch

cv3d opened the pull request:
DO NOT MERGE! This PR is for demonstration purposes only..

### This pullrequest changes
ElemType patch with zero changes from `CV_8U` to `CV_8UC1` etc

cv3d commented on modules/core/include/opencv2/core/affine.hpp:
```diff
@@ -300,23 +300,22 @@ namespace cv
-               ,depth        = DataType<channel_type>::depth
-               ,type         = CV_MAKETYPE(depth, channels)
+        static const ElemType   depth        = DataType<channel_type>::depth;
+        static const ElemType   type         = CV_MAKETYPE(depth, channels);
```
I think i pointed to that in some previous comment! His request boils down to the very core of the reason why `type()` and `.depth()` exists...
I hope he can realize that it is just non-sense to make depth and type the same.

vpisarev commented:
thanks! the goal of this exercise is to figure out what could be the _minimal_ damage in the case the patch is accepted. So, let's reduce the patch as much as possible. Please, see the further comments to the patch

vpisarev commented on modules/core/src/arithm.cpp:
```diff
@@ -61,7 +61,7 @@ void convertAndUnrollScalar( const Mat& sc, int buftype, uchar* scbuf, size_t bl
     int scn = (int)sc.total(), cn = CV_MAT_CN(buftype);
     size_t esz = CV_ELEM_SIZE(buftype);
-    getConvertFunc(sc.depth(), buftype)(sc.ptr(), 1, 0, 1, scbuf, 1, Size(std::min(cn, scn), 1), 0);
+    getConvertFunc(sc.depth(), CV_MAT_DEPTH(buftype))(sc.ptr(), 1, 0, 1, scbuf, 1, Size(std::min(cn, scn), 1), 0);
```
since getConvertFunc() takes 2 ElemType's, let's insert 2 `CV_MAT_DEPTH()` calls into that function instead of using `CV_MAT_DEPTH()` each time we call it.

vpisarev commented on modules/core/src/arithm.cpp:
```diff
@@ -596,16 +605,21 @@ static bool ocl_arithm_op(InputArray _src1, InputArray _src2, OutputArray _dst,
 
 static void arithm_op(InputArray _src1, InputArray _src2, OutputArray _dst,
-                      InputArray _mask, int dtype, BinaryFuncC* tab, bool muldiv=false,
+                      InputArray _mask, ElemType ddepth, BinaryFuncC* tab, bool muldiv=false,
```
please, do not rename arguments; just change the type from int to ElemType

vpisarev commented on modules/core/src/arithm.cpp:
```diff
@@ -615,14 +629,14 @@ static void arithm_op(InputArray _src1, InputArray _src2, OutputArray _dst,
         CV_OCL_RUN(use_opencl,
             ocl_arithm_op(*psrc1, *psrc2, _dst, _mask,
-                          (!usrdata ? type1 : std::max(depth1, CV_32F)),
+                          (!usrdata ? type1 : CV_MAKE_TYPE(CV_MAX_DEPTH(depth1, CV_32F), 1)),
```
CV_MAKE_TYPE is unnecessary here

vpisarev commented on modules/core/src/arithm.cpp:
```diff
@@ -919,26 +928,26 @@ static BinaryFuncC* getAbsDiffTab()
 
 void cv::add( InputArray src1, InputArray src2, OutputArray dst,
-          InputArray mask, int dtype )
+          InputArray mask, ElemType ddepth )
```
restore the original parameter name

vpisarev commented on modules/core/src/arithm.cpp:
```diff
@@ -1658,7 +1671,7 @@ static bool ocl_inRange( InputArray _src, InputArray _lowerb,
             sccvtfunc(lscalar.ptr(), 1, 0, 1, (uchar*)ilbuf, 1, Size(cn, 1), 0);
             sccvtfunc(uscalar.ptr(), 1, 0, 1, (uchar*)iubuf, 1, Size(cn, 1), 0);
-            int minval = cvRound(getMinVal(sdepth)), maxval = cvRound(getMaxVal(sdepth));
+                int minval = cvRound(getMinVal(sdepth)), maxval = cvRound(getMaxVal(sdepth));
```
extra indentation

vpisarev commented on modules/core/src/arithm.cpp:
```diff
@@ -1904,7 +1918,7 @@ CV_IMPL void cvAdd( const CvArr* srcarr1, const CvArr* srcarr2, CvArr* dstarr, c
     if( maskarr )
         mask = cv::cvarrToMat(maskarr);
-    cv::add( src1, src2, dst, mask, dst.type() );
+    cv::add( src1, src2, dst, mask, dst.depth() );
```
restore `.type()`

vpisarev commented on modules/core/src/array.cpp:
```diff
@@ -94,7 +93,7 @@ cvSetIPLAllocators( Cv_iplCreateImageHeader createHeader,
 // Creates CvMat and underlying data
 CV_IMPL CvMat*
-cvCreateMat( int height, int width, int type )
+cvCreateMat(int height, int width, int type)
```
unnecessary whitespace change

vpisarev commented on modules/core/src/convert_scale.cpp:
```diff
@@ -447,28 +449,30 @@ namespace cv {
 #ifdef HAVE_OPENCL
 
-static bool ocl_normalize( InputArray _src, InputOutputArray _dst, InputArray _mask, int dtype,
+static bool ocl_normalize( InputArray _src, InputOutputArray _dst, InputArray _mask, ElemType ddepth,
```
undo `dtype` to `ddepth` rename

vpisarev commented on modules/core/src/cuda_gpu_mat.cpp:
```diff
@@ -494,18 +494,18 @@ GpuMat& cv::cuda::GpuMat::setTo(Scalar s, InputArray _mask, Stream& _stream)
 {
     CV_UNUSED(_dst);
-    CV_UNUSED(rtype);
+    CV_UNUSED(ddepth);
```
undo argument rename

cv3d commented:
@vpisarev Could you please reopen this PR to allow me addressing your concerns?
It seems I did not receive email updates about it for some reason or another..

cv3d commented:
@vpisarev A kind reminder

vpisarev commented:
@cv3d, let's get back to your PRs after OpenCV 4.0 release
